nixos/modules/system/boot/resolved.nix

```
@@ -16,6 +16,7 @@ let
     ${optionalString (cfg.domains != [])
       "Domains=${concatStringsSep " " cfg.domains}"}
     LLMNR=${cfg.llmnr}
+    DNS=${cfg.dns}
```

I'm not sure that it will work as is, how about this instead?
```suggestion
    ${optionalString (cfg.dns != [])
      "DNS=${concatStringsSep " " cfg.dns}"}
```

---

Implemented the change, thanks for catching it.